plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -111,6 +113,27 @@
   public static final String SPARK_NODE_LABELING_JOBTYPE_PROPERTY = "spark.node.labeling.enforced";
   // HadoopSecureSparkWrapper ENV VAR if spark.node.labeling.enforced is set to true
   public static final String SPARK_NODE_LABELING_ENV_VAR = "SPARK_NODE_LABELING_ENFORCED";
+  // Jobtype property for whether to enable auto node labeling for Spark applications
+  // submitted via the Spark jobtype.
+  public static final String SPARK_AUTO_NODE_LABELING_JOBTYPE_PROPERTY = "spark.auto.node.labeling.enabled";
+  // Env var to be passed to {@HadoopSecureSparkWrapper} for whether auto node labeling
+  // is enabled
+  public static final String SPARK_AUTO_NODE_LABELING_ENV_VAR = "SPARK_AUTO_NODE_LABELING_ENABLED";
+  // Jobtype property to configure the desired node label expression when auto node
+  // labeling is enabled and min mem/vcore ratio is met.
+  public static final String SPARK_DESIRED_NODE_LABEL_JOBTYPE_PROPERTY = "spark.desired.node.label";
+  // Env var to be passed to {@HadoopSecureSparkWrapper} for the desired node label expression
+  public static final String SPARK_DESIRED_NODE_LABEL_ENV_VAR = "SPARK_DESIRED_NODE_LABEL";
+  // Jobtype property to configure the minimum mem/vcore ratio for a Spark application's
+  // executor to be submitted with the desired node label expression.
+  public static final String SPARK_MIN_MEM_VCORE_RATIO_JOBTYPE_PROPERTY = "spark.min.mem.vore.ratio";
+  // Env var to be passed to {@HadoopSecureSparkWrapper} for the value of minimum
+  // mem/vcore ratio
+  public static final String SPARK_MIN_MEM_VCORE_RATIO_ENV_VAR = "SPARK_MIN_MEM_VCORE_RATIO";
+  // Jobtype property to define where default spark-defaults.conf file is
+  public static final String SPARK_PROEPRTY_FILE_PATH_JOBTYPE_PROPERTY = "spark.property.file";
```

Comment:
This is used for constructing the SparkConf object. It is different from Hadoop's Configuration object where you can explicitly ask it to load config files on classpath. You need to pass the config file  in order to load the SparkCond object with default configuration.